replication/parser.go

```diff
@@ -106,42 +106,40 @@ func (p *BinlogParser) parseSingleEvent(r io.Reader, onEvent OnEventFunc) (bool,
 	var err error
 	var n int64
 
-	headBuf := make([]byte, EventHeaderSize)
-
-	if _, err = io.ReadFull(r, headBuf); err == io.EOF {
+	var buf bytes.Buffer
+	if n, err = io.CopyN(&buf, r, EventHeaderSize); err == io.EOF {
 		return true, nil
 	} else if err != nil {
-		return false, errors.Trace(err)
+		return false, errors.Errorf("get event header err %v, need %d but got %d", err, EventHeaderSize, n)
 	}
 
 	var h *EventHeader
-	h, err = p.parseHeader(headBuf)
+	h, err = p.parseHeader(buf.Bytes())
 	if err != nil {
 		return false, errors.Trace(err)
 	}
 
 	if h.EventSize <= uint32(EventHeaderSize) {
 		return false, errors.Errorf("invalid event header, event size is %d, too small", h.EventSize)
 	}
-
-	var buf bytes.Buffer
-	if n, err = io.CopyN(&buf, r, int64(h.EventSize)-int64(EventHeaderSize)); err != nil {
-		return false, errors.Errorf("get event body err %v, need %d - %d, but got %d", err, h.EventSize, EventHeaderSize, n)
+	if n, err = io.CopyN(&buf, r, int64(h.EventSize-EventHeaderSize)); err != nil {
+		return false, errors.Errorf("get event err %v, need %d but got %d", err, h.EventSize, n)
+	}
+	if buf.Len() != int(h.EventSize) {
+		return false, errors.Errorf("invalid raw data size in event %s, need %d but got %d", h.EventType, h.EventSize, buf.Len())
 	}
 
-	data := buf.Bytes()
-	rawData := data
-
-	eventLen := int(h.EventSize) - EventHeaderSize
-
-	if len(data) != eventLen {
-		return false, errors.Errorf("invalid data size %d in event %s, less event length %d", len(data), h.EventType, eventLen)
+	rawData := buf.Bytes()
```

Comment:
now the rawdata contains all data, please change the comment for the rawData in the structure.

Reply:
done